### PR TITLE
Remove static Instance of UpdateManager

### PR DESCRIPTION
**Pull request by sharadkjaiswal**
- UpdateManager class is no more a singleton class, hence static property "Instance" is removed.
- DynamoModel holds IUpdateManager instance, which is initialized via StartConfiguration parameter. If StartConfiguration doesn't have a valid instance of IUpdateManager, then it creates an instance of UpdateManager with default configuration parameter.
- Clients can set IUpdateManager while initializing DynamoModel, thru StartConfiguration parameter.
- A static method GetProductVersion() is introduced to return product version so that this can be called even when DynamoModel is not initialized.
- A new property IsUpdateAvailable is defined to return true only if update info is downloaded, and we have a valid update or force update is set true.
- The green cloud on UI is bound to this property, so the cloud should be visible only after data has been downloaded.
- Tests have been updated to make use of UpdateManager and some new tests are added to test IsUpdateAvailable property.

**PTAL**
- [ ] @aparajit-pratap 
- [ ] @Benglin  
- [ ] @ikeough

### CODE REVIEW

File: src/DynamoCore/Models/DynamoModel.cs [hunk 1]
```diff
@@ -524,9 +529,11 @@ protected DynamoModel(IStartConfiguration config)
 
             AddHomeWorkspace();
 
+            UpdateManager = config.UpdateManager ?? new DefaultUpdateManager(null);
+            UpdateManager.Log += updateManager_Log;
```
Comment (aparajit-pratap) [inline src/DynamoCore/Models/DynamoModel.cs]: Please rename the event handler to start with caps.

File: src/DynamoCore/UpdateManager/UpdateManager.cs [hunk 2]
```diff
@@ -434,6 +447,18 @@ public IAppVersionInfo DownloadedUpdateInfo
...
 
+        public bool IsUpdateAvailable
+        {
+            get
+            {
+                //Update is not available unitl it's downloaded
+                if(DownloadedUpdateInfo==null)
+                    return false;
+
+                return ForceUpdate || AvailableVersion > ProductVersion;
```
Comment (aparajit-pratap) [inline src/DynamoCore/UpdateManager/UpdateManager.cs]: Are these properties set by the `CheckForProductUpdate` method?

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs [hunk 3]
```diff
@@ -575,14 +571,14 @@ private void UnsubscribeLoggerEvents()
 
         private void SubscribeUpdateManagerHandlers()
         {
-            UpdateManager.UpdateManager.Instance.UpdateDownloaded += Instance_UpdateDownloaded;
-            UpdateManager.UpdateManager.Instance.ShutdownRequested += updateManager_ShutdownRequested;
+            model.UpdateManager.UpdateDownloaded += Instance_UpdateDownloaded;
+            model.UpdateManager.ShutdownRequested += updateManager_ShutdownRequested;
```
Comment (aparajit-pratap) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs]: rename

File: src/DynamoCoreWpf/Views/About/AboutWindow.xaml [hunk 4]
```diff
@@ -62,7 +62,7 @@
                                PreviewMouseDown="OnPreviewMouseDown"
                                HorizontalAlignment="Center">
                         <Run Foreground="{Binding IsUpdateAvailable, Converter={StaticResource IsUpdateAvailableBrushConverter}}" 
-                             Text="{Binding IsUpdateAvailable, Converter={StaticResource IsUpdateAvailableToTextConverter}, 
+                             Text="{Binding Model.UpdateManager, Converter={StaticResource IsUpdateAvailableToTextConverter},
```
Comment (aparajit-pratap) [inline src/DynamoCoreWpf/Views/About/AboutWindow.xaml]: Can we not bind this to the `IsUpdateAvailable` property on `UpdateManager` directly? Why do we need the converter?

File: src/DynamoCore/UpdateManager/UpdateManager.cs (hunk 2 above)
Comment (sharadkjaiswal) [inline src/DynamoCore/UpdateManager/UpdateManager.cs]: Yes, except for ForceUpdate is set by the configuration or by directly updating the property from Debug UI. Just wanted to ensure that green cloud is visible only after we have DownloadedUpdateInfo

File: src/DynamoCoreWpf/Views/About/AboutWindow.xaml (hunk 4 above)
Comment (sharadkjaiswal) [inline src/DynamoCoreWpf/Views/About/AboutWindow.xaml]: Converter needs to get the available version to show in about box dialog, hence we need to bind with UpdateManager

**Comment by sharadkjaiswal** [comment]
Thanks @aparajit-pratap for your review comments, I have addressed them and going ahead to merge it.

**Pull request merged**